pkg/manifest/manifest.go

```diff
@@ -98,6 +110,69 @@ func (m *Manifest) UnmarshalJSON(in []byte) error {
 	return validateResourceId(m.id)
 }
 
+func (m *Manifest) Include(excludeIdentifier *string, includeTechPreview bool, profile *string, capabilities *ClusterCapabilities) error {
+
+	annotations := m.Obj.GetAnnotations()
+	if annotations == nil {
+		return fmt.Errorf("no annotations")
+	}
+
+	excludeAnnotation := fmt.Sprintf("exclude.release.openshift.io/%s", *excludeIdentifier)
+	if v := annotations[excludeAnnotation]; v == "true" {
+		return fmt.Errorf("%s=%s", excludeAnnotation, v)
+	}
+
+	featureGateAnnotation := "release.openshift.io/feature-gate"
+	featureGateAnnotationValue, featureGateAnnotationExists := annotations[featureGateAnnotation]
+	if featureGateAnnotationValue == string(configv1.TechPreviewNoUpgrade) && !includeTechPreview {
+		return fmt.Errorf("tech-preview excluded, and %s=%s", featureGateAnnotation, featureGateAnnotationValue)
+	}
+	// never include the manifest if the feature-gate annotation is outside of allowed values (only TechPreviewNoUpgrade is currently allowed)
+	if featureGateAnnotationExists && featureGateAnnotationValue != string(configv1.TechPreviewNoUpgrade) {
+		return fmt.Errorf("unrecognized value %s=%s", featureGateAnnotation, featureGateAnnotationValue)
+	}
+
+	profileAnnotation := fmt.Sprintf("include.release.openshift.io/%s", *profile)
+	if val, ok := annotations[profileAnnotation]; ok && val != "true" {
+		return fmt.Errorf("unrecognized value %s=%s", profileAnnotation, val)
+	} else if !ok {
+		return fmt.Errorf("%s unset", profileAnnotation)
+	}
+
+	return checkResourceEnablement(annotations, capabilities)
+}
+
+// CheckResourceEnablement, given resource annotations and defined cluster capabilities, checks if the capability
```

Comment:
nit: `checkResourceEnablement` , to match the now-internal function name

Reply:
fixed in https://github.com/openshift/library-go/commit/f529590c5da24451b978dd972aade0a4bc3d40bd